cdap-app-templates/cdap-etl/cdap-etl-api/src/main/java/io/cdap/cdap/etl/api/engine/SQLOperationResult.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright © 2021 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.cdap.etl.api.engine;
+
+/**
+ * Represents the result of a SQL Engine operation.
+ */
+public class SQLOperationResult {
+  private final SQLOperationStatus status;
+  private final String tableName;
+  private final long numAffectedRecords;
+
+  public SQLOperationResult(SQLOperationStatus status,
+                            String tableName,
+                            long numAffectedRecords) {
+    this.status = status;
+    this.tableName = tableName;
+    this.numAffectedRecords = numAffectedRecords;
+  }
+
+  public SQLOperationStatus getStatus() {
+    return status;
+  }
+
+  public String getTableName() {
+    return tableName;
+  }
+
+  public long getNumAffectedRecords() {
```

Comment:
What is this number? Is it number of records read or produced? I believe currently our metrics has both separate

Reply:
It depends on the operation. For Push/Pull is the number of inserted/fetched rows, for aggregation operations (such as a Join) is the number of rows returned by the join operation.